src/Goerli-DssSpell.sol

```diff
@@ -16,29 +16,62 @@
 
 pragma solidity 0.6.12;
 // Enable ABIEncoderV2 when onboarding collateral through `DssExecLib.addNewCollateral()`
-// pragma experimental ABIEncoderV2;
+pragma experimental ABIEncoderV2;
 
 import "dss-exec-lib/DssExec.sol";
 import "dss-exec-lib/DssAction.sol";
 
 import { DssSpellCollateralAction } from "./Goerli-DssSpellCollateral.sol";
 
-interface RwaUrnLike {
-    function draw(uint256) external;
+import { VatAbstract } from "dss-interfaces/dss/VatAbstract.sol";
+import { JugAbstract } from "dss-interfaces/dss/JugAbstract.sol";
+
+interface CureAbstract {
+    function lift(address) external;
+}
+
+interface TeleportJoinLike {
+    function file(bytes32,bytes32,address) external;
+    function file(bytes32,bytes32,uint256) external;
+}
+
+interface TeleportRouterLike {
+    function file(bytes32,bytes32,address) external;
+}
+
+interface TeleportOracleAuthLike {
+    function file(bytes32,uint256) external;
+    function addSigners(address[] calldata) external;
+}
+
+interface EscrowLike {
+    function approve(address,address,uint256) external;
 }
 
 contract DssSpellAction is DssAction, DssSpellCollateralAction {
 
     // Provides a descriptive tag for bot consumption
     string public constant override description = "Goerli Spell";
 
-    uint256 constant internal MILLION  = 10 **  6;
-    uint256 constant internal BILLION  = 10 **  9;
-    uint256 constant internal WAD      = 10 ** 18;
+    address internal constant TELEPORT_JOIN = 0xAaEf3f1523291aDdaF98a8535c27F25971d823D2;
+    address internal constant ORACLE_AUTH = 0xD40ab915cB8232E8188e1a9137E4b5dCB86F0fd8;
+    address internal constant ROUTER = 0x75c2D0A33cB245acD1F10798fEdD4a42Be4951a9;
+    address internal constant LINEAR_FEE = 0x72Cb460888D401f991AB1a78ffc48EFcDcd155e8;
```

Comment:
Not sure, but should we add this to the chainlog as well? might be something worth tracking.

I know it's all immutable atm, but future versions might have wards that we'll benefit from the auto wards checks that traverse the chainlog.

Reply:
Yeah IMO `LINEAR_FEE` should be in the chainlog with the proper name that corresponds. It is pretty similar to the clippers cals that we are logging.
So we should probably have a key per domain, and we should think if we want to duplicate the contract even if they are the same.

Reply:
I was debating this. We don't necessarily track everything, but it could be added as `OPTIMISM_TELEPORT_FEE` or something.

Reply:
added